test/tests/functional/pbs_soft_walltime.py

```diff
@@ -687,3 +711,323 @@ def test_preempt_order(self):
+            self.assertTrue(msg in e.msg[0])
+
+        try:
+            self.server.manager(MGR_CMD_SET, SERVER, {
+                                'Resources_default.soft_walltime':
+                                '1000000000000000000000000'})
+        except PbsManagerError as e:
+            self.assertTrue(msg in e.msg[0])
+
+        try:
+            self.server.manager(
+                MGR_CMD_SET, SERVER,
+                {'Resources_default.soft_walltime': '-1'})
+        except PbsManagerError as e:
+            self.assertTrue(msg in e.msg[0])
+
+        self.server.manager(
+            MGR_CMD_SET, SERVER,
+            {'Resources_default.soft_walltime': '00:01:00'})
+
+    def test_soft_runjob_hook(self):
+        """
+        Test that soft walltime is set by runjob hook
+        """
+
+        hook_body = \
+            """import pbs
+e = pbs.event()
+e.job.Resource_List["soft_walltime"] = pbs.duration(5)
+e.accept()
+"""
+        a = {'event': 'runjob', 'enabled': 'True'}
+        self.server.create_import_hook("que", a, hook_body)
+
+        J = Job(TEST_USER)
+        jid = self.server.submit(J)
+
+        self.server.expect(JOB, {'Resource_List.soft_walltime': 5}, id=jid)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+
+    def test_soft_modifyjob_hook(self):
+        """
+        Test that soft walltime is set by modifyjob hook
+        """
+
+        hook_body = \
+            """import pbs
+e = pbs.event()
+e.job.Resource_List["soft_walltime"] = pbs.duration(15)
+e.accept()
+"""
+        a = {'event': 'modifyjob', 'enabled': 'True'}
+        self.server.create_import_hook("que", a, hook_body)
+
+        J = Job(TEST_USER)
+        jid = self.server.submit(J)
+
+        self.server.expect(
+            JOB, 'Resource_List.soft_walltime', op=UNSET, id=jid)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+
+        self.server.alterjob(jid, {'Resource_List.soft_walltime': 5})
+        self.server.expect(JOB, {'Resource_List.soft_walltime': 15}, id=jid)
+
+    def test_walltime_default(self):
+        """
+        Test soft walltime behavior with hard walltime is same
+        even if set under resource_default
+        """
+
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {'Resources_default.soft_walltime': '15'})
+
+        J = Job(TEST_USER, attrs={'Resource_List.walltime': 15})
+        jid = self.server.submit(J)
+        self.server.expect(JOB, {'Resource_List.soft_walltime': 15}, id=jid)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+
+        self.server.deljob(jid, wait=True)
+
+        J = Job(TEST_USER, attrs={'Resource_List.walltime': 16})
+        jid1 = self.server.submit(J)
+        self.server.expect(JOB, {'Resource_List.soft_walltime': 15}, id=jid1)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
+        self.server.deljob(jid1, wait=True)
+
+        # following piece is commented due to PP-1058
```

Comment:
Checking in commented code, I'm not sure if that's a good idea, I have a strong feeling that it'll just get forgotten and might not get uncommented post PP-1058. Maybe create a ticket to add this code in after PP-1058 is checked in? It might be more traceable that way.

Reply:
Hey @agrawalravi90 , for now I added a note in the PP ticket. Since this test doesn't exist yet we can log a bug later and add the dependency. what do you suggest.